### PR TITLE
OpenLibrary implementation and search endpoint for books

### PR DESCRIPTION
**Pull request by marcegl91**
### Summary

- Add endpoint to query OpenLibrary service for searching books by ISBN.

### Trello Card

https://trello.com/c/n4N5jXtm/35-integraci%C3%B3n-api-externa

### CODE REVIEW

File: wbooks-api/app/controllers/api/v1/books_controller.rb [hunk 1]
```diff
@@ -10,6 +10,17 @@ def show
       rescue ActiveRecord::RecordNotFound
         head :not_found
       end
+
+      def search
+        book = OpenLibrary.search(params[:isbn])
+
+        if book.empty?
+          head :not_found
```
Comment (floresv) [inline wbooks-api/app/controllers/api/v1/books_controller.rb]: I thinks tihis is better in one line.

File: wbooks-api/app/services/open_library.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+class OpenLibrary
+  def self.valid_isbn?(isbn)
+    (isbn.length == 10 || isbn.length == 13) && isbn.scan(/\D/).empty?
+  end
+
+  def self.search(isbn)
+    return {} unless valid_isbn?(isbn)
+
+    response = HTTParty.get("https://openlibrary.org/api/books?bibkeys=ISBN:#{isbn}"\
```
Comment (floresv) [inline wbooks-api/app/services/open_library.rb]: Freeze string

https://freelancing-gods.com/2017/07/27/an-introduction-to-frozen-string-literals.html

File: wbooks-api/app/services/open_library.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+                            '&format=json&jscmd=data').values
+
+    return {} if response.empty?
+
+    response = response[0]
+
+    { isbn: isbn,
```
Comment (facucompe) [inline wbooks-api/app/services/open_library.rb]: In the projects we usually create a class that represent the response. It's ok to leave it like this here

File: wbooks-api/config/routes.rb [hunk 4]
```diff
@@ -4,7 +4,12 @@
 
   namespace :api do
     namespace :v1 do
-      resources :books, only: [:show, :index]
+      resources :books, only: [:show, :index] do
```
Comment (facucompe) [inline wbooks-api/config/routes.rb:7]: Can you do: 
```
resources :books, only: [:show, :index] do
  member do
    get :search
  end
end
```

File: wbooks-api/app/services/open_library.rb (hunk 2 above)
Comment (facucompe) [reply wbooks-api/app/services/open_library.rb]: I would add an environment variable for `https://openlibrary.org/api`

**Review by facucompe — APPROVED** [review]

**Review by floresv — APPROVED** [review]
LGTM

**Pull request merged**